wsmaster/codenvy-hosted-api-organization/src/main/java/com/codenvy/organization/spi/jpa/JpaMemberDao.java

```diff
@@ -0,0 +1,197 @@
+        try {
+            final EntityManager manager = managerProvider.get();
+            return manager.createNamedQuery("Member.getByOrganization", MemberImpl.class)
+                          .setParameter("organizationId", organizationId)
+                          .getResultList();
+        } catch (RuntimeException e) {
+            throw new ServerException(e.getLocalizedMessage(), e);
+        }
+    }
+
+    @Override
+    @Transactional
+    public List<MemberImpl> getMemberships(String userId) throws ServerException {
+        requireNonNull(userId, "Required non-null user id");
+        try {
+            final EntityManager manager = managerProvider.get();
+            return manager.createNamedQuery("Member.getByUser", MemberImpl.class)
+                          .setParameter("userId", userId)
+                          .getResultList();
+        } catch (RuntimeException e) {
+            throw new ServerException(e.getLocalizedMessage(), e);
+        }
+    }
+
+    @Override
+    @Transactional
+    public Page<OrganizationImpl> getOrganizations(String userId, int maxItems, int skipCount) throws ServerException {
+        requireNonNull(userId, "Required non-null user id");
+        checkArgument(maxItems >= 0, "The number of items to return can't be negative.");
+        checkArgument(skipCount >= 0, "The number of items to skip can't be negative.");
+        try {
+            final EntityManager manager = managerProvider.get();
+            final List<OrganizationImpl> result = manager.createNamedQuery("Member.getOrganizations", OrganizationImpl.class)
+                                                         .setParameter("userId", userId)
+                                                         .setMaxResults(maxItems)
+                                                         .setFirstResult(skipCount)
+                                                         .getResultList();
+            final Long organizationsCount = manager.createNamedQuery("Member.getOrganizationsCount", Long.class)
+                                                   .setParameter("userId", userId)
+                                                   .getSingleResult();
+
+            return new Page<>(result, skipCount, maxItems, organizationsCount);
+        } catch (RuntimeException e) {
+            throw new ServerException(e.getLocalizedMessage(), e);
+        }
+    }
+
+    @Transactional
+    protected void doRemove(String organizationId, String userId) {
+        final EntityManager manager = managerProvider.get();
+        try {
+            MemberImpl member = manager.createNamedQuery("Member.getMember", MemberImpl.class)
+                                       .setParameter("userId", userId)
+                                       .setParameter("organizationId", organizationId)
+                                       .getSingleResult();
+            manager.remove(member);
+        } catch (NoResultException ignored) {
```

Comment:
you can avoid `ignored` exception by requesting result list, i like this approach more
```
List<MemberImpl> members = (...).getResultList();
if (!members.isEmpty()) {
    manager.remove(members.get(0));
}
````